model/event/build_event.go

```diff
@@ -0,0 +1,39 @@
+package event
+
+import (
+	"time"
+
+	"github.com/mongodb/grip"
+)
+
+func init() {
+	registry.AddType(ResourceTypeBuild, buildEventDataFactory)
+	registry.AllowSubscription(ResourceTypeBuild, BuildStateChange)
+}
+
+const (
+	ResourceTypeBuild = "BUILD"
+
+	BuildStateChange = "STATE_CHANGE"
+)
+
+type BuildEventData struct {
+	Status string `bson:"status" json:"string"`
+}
+
+func LogBuildStateChangeEvent(id, status string) {
```

Comment:
I think it would be better for this to return an error and let the caller handle it

Reply:
There is nothing the caller can do. They are not going to stop whatever processing they're doing if this operation fails. The most they can do is log it, and move on. Thus, I'm inclined to keep doing this.

I've made EVG-3392 to add structure to all the event logs.